xknx/management/management.py

```diff
@@ -359,5 +367,15 @@ async def request(self, payload: APCI, expected: type[APCI] | None) -> Telegram:
             raise ManagementConnectionRefused(
                 "Management connection disconnected by the peer."
             )
+
+        if self.rate_limit:
+            # time in seconds since the last request operation
+            time_diff = time.time() - self._last_sent_time
+            wait_time = 1 / self.rate_limit
+            if time_diff < wait_time:
+                await asyncio.sleep(wait_time - time_diff)
+
         await self._send_data(payload)
-        return await self._receive(expected)
+        response = await self._receive(expected)
+        self._last_sent_time = time.time()
```

Comment:
```suggestion
        self._last_sent_time = time.time()
        response = await self._receive(expected)
```
The current implementation stores the "`_last_response_time`", not the time of sending the last payload. I'm not sure if the variable name is misleading or you actually want to rate limit only on sending.

Reply:
I failed to rename this variable. The idea was indeed to store the last response time as the whole exchange is considered.